CODEOWNERS: clean-up entries for deleted files

qmonnet opened the pull request:
Remove from `CODEOWNERS` the patterns for which we no longer have any entry in the repository. List obtained with:

```bash
while read i; do
    case "$i" in
        /*)
            # Remove leading slash and use ls
            LIST+=" ${i#/}"
            ;;
        *)
            # No leading slash: maybe not at the root, use find
            [[ -n $(find . -name "$i" -print -quit) ]] || echo "$i"
            ;;
    esac
done <<< $(awk '/^[^#]/ {print $1}' CODEOWNERS)
ls -- $LIST 2>&1 >/dev/null | sed "s=.*'\(.*\)':.*=/\1="

```

The above script may be added to a GitHub action as a follow-up, to keep the file up-to-date, if there is any interest.

Note: I did not check that all files in the repo are covered by `CODEOWNERS` because 1) it's more complex (need to filter-out files ignored by Git etc.), and 2) they _are_ all covered by the generic `*` line anyway.

qmonnet committed:
CODEOWNERS: clean-up entries for deleted files

Remove from CODEOWNERS the patterns for which we no longer have any
entry in the repository. List obtained with:

    while read i; do
        case "$i" in
            /*)
                # Remove leading slash and use ls
                LIST+=" ${i#/}"
                ;;
            *)
                # No leading slash: maybe not at the root, use find
                [[ -n $(find . -name "$i" -print -quit) ]] || echo "$i"
                ;;
        esac
    done <<< $(awk '/^[^#]/ {print $1}' CODEOWNERS)
    ls -- $LIST 2>&1 >/dev/null | sed "s=.*'\(.*\)':.*=/\1="

Fixes: b8401aa2edd6 ("checkpatch: remove checkpatch-related files from the repository")
Fixes: 72e7740245c5 ("doc: Move goverance documentation to a more visible")
Fixes: bf6039b99c33 ("doc: Remove obsolete Docker getting started guide")
Fixes: db9b6f71453c ("docs: add cilium-operator technical overview documentation")
Fixes: 26a80c381696 ("jenkinsfile: Remove stale symlinks")
Fixes: b667f010fb81 ("pkg/k8s: self contain CRDs in common directory")
Signed-off-by: Quentin Monnet <quentin@isovalent.com>

tklauser commented on CODEOWNERS:
```diff
@@ -83,9 +81,7 @@ Makefile* @cilium/build
 /Documentation/check-helmvalues.sh @cilium/docs-structure
-/Documentation/cilium_operator.rst @cilium/operator
 /Documentation/cmdref/ @cilium/nonexistantteam
-/Documentation/commit-access.rst @cilium/contributing @cilium/docs-structure
```
This was renamed in 72e7740245c59832faaf85d24bcc4c10539e3882 to `Documentation/contributing/governance/commit_access.rst`, so probably worth to rename here instead of deleting.

tklauser commented on CODEOWNERS:
```diff
@@ -83,9 +81,7 @@ Makefile* @cilium/build
 /Documentation/check-crd-compat-table.sh @cilium/docs-structure
 /Documentation/check-examples.sh @cilium/docs-structure
 /Documentation/check-helmvalues.sh @cilium/docs-structure
-/Documentation/cilium_operator.rst @cilium/operator
```
This was renamed to `Documentation/internals/cilium_operator.rst` in db9b6f71453cdb99253ce22a586d0a7e5b66322e, so probably better to change the file path here instead of deleting it.

tklauser commented on CODEOWNERS:
```diff
@@ -115,7 +110,6 @@ Makefile* @cilium/build
 /Documentation/gettingstarted/kubeproxy-free.rst @cilium/loadbalancer @cilium/docs-structure
 /Documentation/gettingstarted/policy-creation.rst @cilium/policy @cilium/docs-structure
 /Documentation/glossary.rst @cilium/docs-structure
-/Documentation/hubble.rst @cilium/hubble @cilium/docs-structure
```
This was renamed to `Documentation/internals/hubble.rst` in db9b6f71453cdb99253ce22a586d0a7e5b66322e, so probably better to change the file path here instead of deleting it.

qmonnet committed:
CODEOWNERS: fix wildcard patterns for files under daemon/cmd/

The syntax for wildcards in the CODEOWNERS file is a simple asterisk
"*", which should not be preceded by a dash (".*") like other languages
may use for regular expressions.

For most entries, this means that only files starting by e.g.
"ipcache.", such as "ipcache.go", are covered, but not
"ipcache_test.go". For "kube_proxy.*", this even means there is no
related entries, given that all files follow the pattern "kube_proxy_*".

Let's replace the use of ".*" by "*" (or simply ".go" where relevant) in
the CODEOWNERS file.

References:
- https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax
- https://git-scm.com/docs/gitignore#_pattern_format

Fixes: a6677888fe04 ("CODEOWNERS: Attach entries to root of repository")
Signed-off-by: Quentin Monnet <quentin@isovalent.com>

tklauser reviewed (approved):
Thanks Quentin!

pchaigno reviewed (approved):
Thanks!

I think it would be worth adding this script as a quick GitHub workflow.
